### PR TITLE
Mysql 8 compliance

### PR DESCRIPTION
**Pull request by axehomeyg**
I needed a few changes to support MySQL 8 databases which included...

1. upgrading the `mysql` dependency to `mysql2` for new Mysql 8 password  encryption compliance (to avoid the following error message)

```
error Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; 
```

2. The `tablename`.`*` syntax doesn't work in mysql 8, so, addressed that issue in the Select compiler.

3. The integration tests (at least on m2 Mac) would fail because of the small sleep duration. Apparently it takes longer than 10 seconds to bootstrap a Mysql container on an M2 Pro!

It's been a very useful library for me in my current project; thanks for building/maintaining it!

### CODE REVIEW

File: src/connection/Database.ts [hunk 1]
```diff
@@ -48,7 +48,9 @@ export default class Database
         return new Promise((resolve, reject) => {
             this.pool.getConnection((error, connection) => {
                 if (error) {
-                    connection.destroy();
+                    console.log("ERROR", error);
```
Comment (matthewwolfe) [inline src/connection/Database.ts]: Could you please remove the console.log here, and add some curly brackets for the if statement below?

```ts
if (connection) {
  connection.destroy();
}
```

**Comment by matthewwolfe** [comment]
Thanks for the PR! I left one minor comment of feedback, but otherwise I'll get this merged and published! I'm surprised to see anyone using this haha, perhaps I should give it a bit more love with some automation and dependency updates as well.

File: src/connection/Database.ts (hunk 1 above)
Comment (axehomeyg) [reply src/connection/Database.ts]: Updated!

**Comment by axehomeyg** [comment]
> Thanks for the PR! I left one minor comment of feedback, but otherwise I'll get this merged and published! I'm surprised to see anyone using this haha, perhaps I should give it a bit more love with some automation and dependency updates as well.

I'm a Rails dude normally, so, I was glad to see an AR-style option for a quick Node project that I'm working on!

**Comment by matthewwolfe** [comment]
2.0.31 has been published

**Pull request merged**